lib/mix/phx_gen_auth/injectors/config.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Mix.Phx.Gen.Auth.Injectors.Config do
+  @moduledoc false
+
+  alias Mix.Phx.Gen.Auth.{HashingLibrary, Injector}
+
+  def inject(file, %HashingLibrary{} = hashing_library) when is_binary(file) do
+    code_to_inject =
+      hashing_library
+      |> code_to_inject()
+      |> normalize_line_endings_to_file(file)
+
+    Injector.inject_unless_contains(
+      file,
+      code_to_inject,
+      # Matches the entire line and captures the line ending. In the
+      # replace string:
+      #
+      # * the entire matching line is inserted with \\0,
+      # * the actual code is injected with &2,
+      # * and the appropriate newlines are injected using \\2.
+      &Regex.replace(~r/(use Mix\.Config|import Config)(\r\n|\n|$)/, &1, "\\0\\2#{&2}\\2", global: false)
+    )
```

Comment:
Btw, I would consider keeping this function in Injector.ex itself. If necessary, use a common prefix or code comments to mark which functions are used together.

Reply:
Thanks for this feedback. :heart:  

Just to double-check my understanding, you're saying this because:

* it allows us to see similar code in the same module
* and we don't have to hunt around to figure out which module injects the config/app menu/router plug 

Regarding tests, because tests for one of these functions can get quite lengthy, would you keep all of these tests in a single injector_test.ex or break out different test files per injection like injector/config_test.exs?

Reply:
> Just to double-check my understanding, you're saying this because:

Exactly.

>  a single injector_test.ex

Yup, and use describe blocks to separate each function being tested.